### PR TITLE
Pipelined api

### PR DESCRIPTION
**Pull request by gbaydin**
This PR is to keep track of the curried/tupled API restructuring discussed in #155

### CODE REVIEW

File: src/DiffSharp.Core/DiffSharp.Compose.fs [hunk 1]
```diff
@@ -0,0 +1,283 @@
+// Pipelined operations
+module DiffSharp.Compose
```
Comment (dsyme) [inline src/DiffSharp.Core/DiffSharp.Compose.fs:1]: Nice name

File: src/DiffSharp.Core/DiffSharp.Shorten.fs [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// Pipelined operations
+module DiffSharp.Shorten
```
Comment (dsyme) [inline src/DiffSharp.Core/DiffSharp.Shorten.fs:1]: Maybe "DiffSharp.Abbreviations" though either is ok

**Review by dsyme — APPROVED** [review]

File: src/DiffSharp.Core/DiffSharp.Shorten.fs (hunk 2 above)
Comment (gbaydin) [reply src/DiffSharp.Core/DiffSharp.Shorten.fs:1]: I've been thinking about many names. At some point it was "Shorthand". Then I shortened the shorthand to this.

File: src/DiffSharp.Core/DiffSharp.Compose.fs (hunk 1 above)
Comment (gbaydin) [reply src/DiffSharp.Core/DiffSharp.Compose.fs:1]: I like it too.

File: src/DiffSharp.Core/DiffSharp.Shorten.fs (hunk 2 above)
Comment (gbaydin) [reply src/DiffSharp.Core/DiffSharp.Shorten.fs:1]: I thought about "Abbreviations" too, but it felt too long. "Abbrev" would not be great either.

**Comment by gbaydin** [comment]
@dsyme I'm going ahead and merging this for 1.0. We can revisit some choices post-1.0, but I'm quite happy with this design and compromise. Let's get this in and I will focus on the other things.

**Pull request merged**